tests/PSRule.Rules.Azure.Tests/Azure.CDN.Tests.ps1

```diff
@@ -118,5 +118,23 @@ Describe 'Azure.CDN' -Tag 'CDN' {
             $ruleResult.Outcome | Should -Be 'Fail';
             # TODO: $ruleResult.Detail.Reason.Path | Should -BeIn 'name';
         }
+        
+        It 'Azure.CDN.UseFrontDoor' {
+            $dataPath = Join-Path -Path $here -ChildPath 'Resources.CDN.json'
+            $result = Invoke-PSRule @invokeParams -InputPath $dataPath
+            $filteredResult = $result | Where-Object { $_.RuleName -eq 'Azure.CDN.UseFrontDoor' };
+
+            # Fail
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 1;
+            $ruleResult.TargetName | Should -BeIn 'frontDoorProfile-A';
+            $ruleResult.Detail.Reason.Path | Should -Be 'sku.name';
+
+            # Pass
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Pass' });
+            $ruleResult | Should -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 0;
```

Comment:
We should have some passing cases for the test. Adding the updates to the rule pre-condition function should help.